cobalt/network/BUILD.gn

```diff
@@ -97,12 +104,12 @@ static_library("network") {
       "net_log_constants.cc",
       "net_log_constants.h",
     ]
-    defines = [ "ENABLE_NETWORK_LOGGING" ]
   }
 
   if (!sb_is_modular) {
     deps += cobalt_platform_dependencies
   }
+  all_dependent_configs = [ ":network_all_dependent_config" ]
```

Comment:
Why does this need to be propagated in the feature branch but not trunk?

Reply:
`ENABLE_NETWORK_LOGGING` needs to be defined when dependents are compiled, which is why I moved it to a config (commit [`2da45b7` (#2612)](https://github.com/youtube/cobalt/pull/2612/commits/2da45b76bcfcf90737bb9cfe0accd7fea381037c)). Specifically for `text_encoding_test` which instantiates a `NetworkModule`, which in this branch has a different ABI due to `CobaltNetLog` being inside an `#if defined(ENABLE_NETWORK_LOGGING)`. In trunk, that is `#if` does not exist.

Reply:
I think this should be in `public_configs` as well to keep dependencies in order and configs scoped

Reply:
Here the same argument as above applies, it's fragile, binaries will compile with mismatching ABIs.

Reply:
TODO added in https://github.com/youtube/cobalt/pull/2612/commits/ee4fd0c9cbdf45bcb6fcba74af9914d36ef0879b